foundation/device/device-id/src/main/kotlin/com/pingidentity/device/id/DefaultDeviceIdentifier.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2025 Ping Identity Corporation. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package com.pingidentity.device.id
+
+import java.security.MessageDigest
+
+/**
+ * The default implementation of [DeviceIdentifier] that uses the Android KeyStore
+ * to generate a persistent device identifier.
+ *
+ * This implementation:
+ * - Creates an RSA key pair in the Android KeyStore the first time it's accessed
+ * - Uses the public key as the basis for a device identifier
+ * - Applies SHA-256 hashing to the key data and converts it to a hex string
+ * - Caches the identifier for efficient subsequent access
+ *
+ * Note: The identifier will change if the device is factory reset or if the key is
+ * explicitly removed from the KeyStore.
+ */
+object DefaultDeviceIdentifier : DeviceIdentifier {
+
+    /** The key alias used to store the device identifier key in the Android KeyStore */
+    internal const val KEY_ALIAS = "com.pingidentity.device.id.DEVICE_IDENTIFIER"
+
+    /**
+     * A lazily initialized device identifier string.
+     * The value is computed once on first access and cached for subsequent calls.
+     *
+     * @return A unique identifier string for the device
+     */
+    override val id: String by lazy { identifier() }
+
+    /**
+     * Generates the device identifier by retrieving or creating a key in the KeyStore,
+     * then hashing the encoded key data.
+     *
+     * @return A SHA-256 hash of the key data as a hex string
+     */
+    @OptIn(ExperimentalStdlibApi::class)
+    private fun identifier(): String {
+        return MessageDigest.getInstance("SHA256")
```

Comment:
Same as on the other file, I think it's worth it using a safe try/catch approach on these to avoid crashes on devices that the keystore is not working as expected

Reply:
The runtime exception will propagate to the caller and eventually be wrapped in a DeviceIdentifierException. However, developers still need to handle the exception to prevent crashes, unless a fallback or default value is provided.